test/integration/integration.cc

```diff
@@ -281,11 +281,11 @@ uint32_t BaseIntegrationTest::lookupPort(const std::string& key) {
 }
 
 void BaseIntegrationTest::registerTestServerPorts(const std::vector<std::string>& port_names) {
-  int index = 0;
-  for (const auto& it : port_names) {
-    Network::ListenSocket* listen_socket = test_server_->server().getListenSocketByIndex(index++);
-    RELEASE_ASSERT(listen_socket != nullptr);
-    registerPort(it, listen_socket->localAddress()->ip()->port());
+  auto port_it = port_names.begin();
+  auto listeners = test_server_->server().listenerManager().listeners();
+  auto listener_it = listeners.begin();
+  for (; port_it != port_names.end() && listener_it != listeners.end(); ++port_it, ++listener_it) {
```

Comment:
Unrelated: I keep wishing there was was a `zip` like operator on iterators for C++.